container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
+		err := errors.New("environment variable sought after does not exist in evnrionemnet map")
+		if err != nil {
+			errors.Wrap(err, "failed to create new error")
+		}
+		return "", err
+	}
+	return os.Expand(value, mapper), nil
+}
+
+func main() {
+	flag.Var(&labelsArray, "labels", "Augment the Label of the previous layer.")
+	flag.Var(&ports, "ports", "Augment the ExposedPorts of the previous layer.")
+	flag.Var(&volumes, "volumes", "Augment the Volumes of the previous layer.")
+	flag.Var(&entrypointPrefix, "entrypointPrefix", "Prefix the Entrypoint with the specified arguments.")
+	flag.Var(&env, "env", "Augment the Env of the previous layer.")
+	flag.Var(&command, "command", "Override the Cmd of the previous layer.")
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layer, "layer", "Layer sha256 hashes that make up this image.")
+	flag.Var(&stampInfoFile, "stampInfoFile", "A list of files from which to read substitutions to make in the provided fields.")
+	flag.Parse()
+	log.Println("Running the Image Config manipulator...")
+
+	if *outputConfig == "" {
+		log.Fatalln("Required option -outputConfig was not specified.")
+	}
+	if *nullEntryPoint == "true" {
+		entrypoint = nil
+	}
+	if *nullCmd == "true" {
+		command = nil
+	}
+
+	// read config file into struct.
+	configPath, err := os.Open(*baseConfig)
+	if err != nil {
+		log.Fatalf("Failed to read the parent image's config file: %v", err)
+	}
+	configFile, err := v1.ParseConfigFile(configPath)
+	if err != nil {
+		log.Fatalf("Failed to successfully parse config file json contents: %v", err)
+	}
+
+	// read manifest file into struct if provided.
+	if *baseManifest != "" {
+		manifestPath, err := ioutil.ReadFile(*baseManifest)
+		if err != nil {
+			log.Fatalf("Failed to read the parent image's manifest: %v", err)
+		}
+		manifestFile := v1.Manifest{}
+		if err = json.Unmarshal([]byte(manifestPath), &manifestFile); err != nil {
+			log.Fatalf("Failed to successfully read manifest file contents: %v", err)
+		}
+	}
+
+	var createTime string
+	if *creationTime == "" {
+		creationTime = nil
+	} else {
+		var unixTime int64
+		if createTime, err = stamp(*creationTime); err != nil {
+			log.Fatalf("Unable to format creation time from BUILD_TIMESTAMP macros: %v", err)
+		}
+		// if creationTime is parsable as a floating point type, assume unix epoch timestamp.
+		// otherwise, assume RFC 3339 date/time format.
+		unixTime, err := strconv.ParseInt(createTime, 10, 64)
+		if err != nil {
+			log.Fatalf("Unable to parse a floating point type from flag creationTime: %v", err)
+		} else {
```

Comment:
This else is unnecessary because the body of the if condition terminates the program.